plugins/woocommerce/includes/class-wc-session-handler.php

```diff
@@ -94,11 +94,26 @@ public function init_session_cookie() {
 			$this->_session_expiration = $cookie[1];
 			$this->_session_expiring   = $cookie[2];
 			$this->_has_cookie         = true;
-			$this->_data               = $this->get_session_data();
+
+			/**
+			 * Allows filtering of the session data retrieved from storage during session cookie initialization.
```

Comment:
How about adding an extra line explaining why stores/exts might want to use this hook?